20181204--Sophia

```diff
@@ -15,8 +15,14 @@ public Phrase(String p)
      */
     public static int findNthOccurrence(String str, int n)
     {
-        /* implementation not shown */
-        int loc = currentPhrase.indexOf(str);
+        int loc = -1;
+        int count = 1;
+        String a = currentPhrase;
+        while(a.indexOf(str)!=-1&&count!=n){
+            loc = currentPhrase.length()-a.length()+a.indexOf(str);
+            a = currentPhrase.substring(loc+1);
+            count++;
+        }
```

Comment:
1. If n is equal to  1, your code will always return -1, even if the 1st occurrence exists.
2. If n is not equal to 1, you will get the index of n-1th occurrence. because you count start from 1
3. even you change loc to 0, count to 0, there is also one big problem. Try to figure it out by yourself.